pkg/testrunner/testrunner.go

```diff
@@ -54,19 +54,35 @@ func (rl RunList) Run(log logr.Logger, tmClient kubernetes.Interface, namespace,
 		go func(i int) {
 			defer wg.Done()
 
-			tr, err := runTestrun(log, tmClient, rl[i].Testrun, namespace, testrunNamePrefix)
-			if err != nil {
-				log.Error(err, "unable to run testrun")
+			for flakeAttempt := 0; flakeAttempt <= maxFlakeAttempts; flakeAttempt++ {
+				tr, err := runTestrun(log, tmClient, rl[i].Testrun, namespace, testrunNamePrefix)
+				if err != nil {
+					log.Error(err, "unable to run testrun")
 
-				if trerrors.IsTimeout(err) {
-					rl[i].Testrun.Status.Phase = tmv1beta1.PhaseStatusTimeout
+					if trerrors.IsTimeout(err) {
+						rl[i].Testrun.Status.Phase = tmv1beta1.PhaseStatusTimeout
+					}
+				}
+				if tr != nil {
+					rl[i].Testrun = tr
+					rl[i].Metadata.Testrun.ID = tr.Name
+				}
+				rl[i].Error = err
+
+				if err == nil && tr.Status.Phase == tmv1beta1.PhaseStatusSuccess {
+					if flakeAttempt != 0 {
+						tr.Status.Flaked = true
+					}
+					// testrun was successful, break retry loop
+					break
+				} else {
```

Comment:
else in not necessary as we break out in the if branch anyways

Reply:
if flakeAttempts > 1, the else branch ist required